dart/lib/src/sentry.dart

```diff
@@ -108,9 +106,28 @@ class Sentry {
     _hub = Hub(options);
     hub.close();
 
-    // execute integrations after hub being enabled
-    for (final integration in options.integrations) {
-      await integration(HubAdapter(), options);
+    if (appRunner != null) {
+      final runZonedGuardedIntegration = options.integrations
+        .whereType<RunZonedGuardedIntegration>().first;
+      assert(
+        runZonedGuardedIntegration != null, 
+        'RunZonedGuardedIntegration should have been added when there is an appRunner.'
+      );
+
+      runZonedGuardedIntegration.runner = () async {
```

Comment:
https://github.com/getsentry/sentry-dart/pull/280/files#r563735742

trying to think a way around this, it looks odd, I'd prefer to create the instance of `RunZonedGuardedIntegration` here if `appRunner` is available, that passes the clojure via ctor